docs/docs-beta/docs/tutorial/pinecone.md

```diff
@@ -0,0 +1,311 @@
+
+One way to handle preprocessing of the data is with [DuckDB](https://duckdb.org/). DuckDB is an in-process database, similar to SQLite, optimized for analytical workloads. We will start by creating two Dagster assets to load in the data. Each will load one of the files and create a DuckDB table (`graphic_novels` and `reviews`):
+```python
+# asssets.py
+@dg.asset(
+    kinds={"duckdb"},
+    group_name="ingestion",
+    deps=[goodreads],
+)
+def graphic_novels(duckdb_resource: dg_duckdb.DuckDBResource):
+    url = "https://datarepo.eng.ucsd.edu/mcauley_group/gdrive/goodreads/byGenre/goodreads_books_comics_graphic.json.gz"
+    query = f"""
+        create table if not exists graphic_novels as (
+          select *
+          from read_json(
+            '{url}',
+            ignore_errors = true
+          )
+        );
+    """
+    with duckdb_resource.get_connection() as conn:
+        conn.execute(query)
+
+@dg.asset(
+    kinds={"duckdb"},
+    group_name="ingestion",
+    deps=[goodreads],
+)
+def reviews(duckdb_resource: dg_duckdb.DuckDBResource):
+    url = "https://datarepo.eng.ucsd.edu/mcauley_group/gdrive/goodreads/byGenre/goodreads_reviews_comics_graphic.json.gz"
+    query = f"""
+        create table if not exists reviews as (
+          select *
+          from read_json(
+            '{url}',
+            ignore_errors = true
+          )
+        );
+    """
+    with duckdb_resource.get_connection() as conn:
+        conn.execute(query)
+```
+
+With our DuckDB tables created, we can now query them like any other SQL table. Our third asset will join and filter the data and then return a DataFrame (we will also `LIMIT` the results to 500):
+```python
+# asssets.py
+@dg.asset(
+    kinds={"duckdb"},
+    group_name="ingestion",
+)
+def enriched_reviews(
+    duckdb_resource: dg_duckdb.DuckDBResource,
+    graphic_novels,
+    reviews,
+) -> pd.DataFrame:
+    query = f"""
+        select
+          reviews.review_id as review_id,
+          graphic_novels.title as title,
+          reviews.rating as rating,
+          reviews.review_text as text
+        from graphic_novels
+        left join reviews
+          on graphic_novels.book_id = reviews.book_id
+        where graphic_novels.language_code = 'eng'
+        limit 500
+    """
+    with duckdb_resource.get_connection() as conn:
+        return conn.execute(query).fetch_df()
+```
+
+Now that the data has been prepared, we are ready to work with our vector database.
+
+### Vector Database
+We can begin by creating the index within our vector database. A vector database is a specialized database designed to store, manage, and retrieve high-dimensional vector embeddings, enabling efficient similarity search and machine learning tasks. There are many different vector databases available. For this demo we will use Pinecone which is a cloud based vector database that offers a [free tier](https://app.pinecone.io/) that can help us get started.
```

Comment:
At this point in the tutorial I had the question, "Why do we need both DuckDB and Pinecone databases?"

I understand that they serve different purposes, but we could maybe clarify more upfront on their usage.